apps/organizations/views.py

```diff
@@ -0,0 +1,86 @@
+# Django Imports
+from django.contrib.auth.decorators import login_required
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from django.utils.decorators import method_decorator
+from django.views import View
+
+# HTK Imports
+from htk.apps.organizations.enums import OrganizationMemberRoles
+from htk.utils import (
+    htk_setting,
+    resolve_model_dynamically,
+)
+from htk.view_helpers import render_custom as _r
+
+
+class OrganizationInvitationResponseView(View):
+    """Organization Invitation Response Class Based View
+
+    This class view intended to be extended in project to set the custom render
+    method and template name there.
+
+    Template should contain a form with POST method and should send `response`
+    variable to this view's POST method.
+
+    `response` value should be either `accept` or `decline`.
+
+    Sample template form:
+    ```
+    <form action="" method="post">
+    <button type="submit" name="response" value="accept">Accept</button>
+    <button type="submit" name="response" value="decline">Decline</button>
+    </form>
+    ```
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.render_method = _r
+        self.template_name = 'organizations/invitation_response.html'
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        token = kwargs.pop('token')
+        OrganizationInvitation = resolve_model_dynamically(
+            htk_setting('HTK_ORGANIZATION_INVITATION_MODEL')
+        )
+        invitation = get_object_or_404(OrganizationInvitation, token=token)
+        is_authenticated = request.user.is_authenticated
+
+        # Logged in user cannot accept invitation in place of someone else.
+        if is_authenticated and request.user.email != invitation.email:
+            raise Http404
```

Comment:
```suggestion
        user = request.user if request.user.is_authenticated else None

        if user and user.email != invitation.email and not(user.profile.has_email(email)):
            raise Http404
```

django-htk supports 1:M user:email

See: https://github.com/hacktoolkit/django-htk/blob/e05822a11d01add512ecd6cc42b92a3e28a8bffd/apps/accounts/models.py#L227

Reply:
Done